src/main/java/org/ibp/api/rest/labelprinting/SubObservationDatasetLabelPrinting.java

```diff
@@ -337,7 +338,12 @@ public LabelsData getLabelsData(final LabelsGeneratorInput labelsGeneratorInput)
 						row.put(requiredField, this.getSeason(observationUnitRow.getVariables().get(field.getName()).getValue()));
 						continue;
 					}
-					row.put(requiredField, observationUnitRow.getVariables().get(field.getName()).getValue());
+					if (observationUnitRow.getVariables().containsKey(field.getName())) {
```

Comment:
Can't we abstract where the value comes from By providing a method in `ObservationUnitRow `class that gets the value where it finds the column name (either from `variables `or `environmentVariables`)?

Reply:
Hi @darla-leafnode, I think that'll work if a variable is expected to be present only in either variables or environmentVariables. But what if they are both present in variables or environmentVariables? Then the code needs to decide where to get the value of based on where a variable is added (VariableType). Just like how it is done here: https://github.com/IntegratedBreedingPlatform/BMSAPI/pull/228#discussion_r313206753

Now for this particular code, there's actually a bug in Create Labels, in which, if a variable is present in both environment and observation level, the system will only display that variable under "Environment" section. As a consequence the variable will only be present in environmentVariables. So right now for this code, we just need to get the variable value either from variables or environmentVariables. This works for now, but this should be fixed soon.